apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/IndyPluginClassLoader.java

```diff
@@ -39,9 +39,14 @@
 
     private static final ClassLoader SYSTEM_CLASS_LOADER = ClassLoader.getSystemClassLoader();
 
+    private final ClassLoader targetClassLoader;
+    private final ClassLoader agentClassLoader;
+
     public IndyPluginClassLoader(@Nullable ClassLoader targetClassLoader, ClassLoader agentClassLoader, Map<String, byte[]> typeDefinitions) {
         // See getResource on why we're using PersistenceHandler.LATENT over PersistenceHandler.MANIFEST
         super(getParent(targetClassLoader, agentClassLoader), true, typeDefinitions, PersistenceHandler.LATENT);
+        this.targetClassLoader = targetClassLoader;
+        this.agentClassLoader = agentClassLoader;
```

Comment:
No need for that, only print the value of `getParent().toString()`. 
In addition, implement a proper `toString()` for `ShadedClassLoader`.
Then, you will get what you want either if the parent is the agent CL, or if it is a `DiscriminatingMultiParentClassLoader`.